aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -115,11 +115,15 @@ void remainder_kernel(TensorIterator& iter) {
     AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "remainder_cpu", [&]() {
       cpu_kernel_vec(iter,
         [=](scalar_t a, scalar_t b) __ubsan_ignore_float_divide_by_zero__ -> scalar_t {
-          return a - b * at::native::floor_impl(a / b);
+          scalar_t mod = std::fmod(a, b);
+          if ((mod != 0) && ((b < 0) != (mod < 0))) mod += b;
+          return mod;
         },
         [=](Vec256<scalar_t> a, Vec256<scalar_t> b) {
-          Vec256<scalar_t> r = a - b * (a / b).floor();
-          return r;
+          auto mod = a.fmod(b);
+          const auto zero = Vec256<scalar_t>(0);
```

Comment:
I noticed that other functions in BinaryOpsKernel.cpp declare constants outside the lambda, e.g. before the `cpu_kernel_vec()` call, but I doubt this has a perf impact...